src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,505 @@ title: Builder.io & Astro
 description: Add content to your Astro project using Builder.io’s visual CMS
 type: cms
 service: Builder.io
-stub: true
+stub: false
 i18nReady: false
 ---
 import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro';
 
+[Builder.io](https://www.builder.io/) is a visual CMS that supports drag-and-drop content editing for building websites. 
 
+At the end of this recipe, you'll be able to create blog posts in the Builder app and have your Astro site update with this new content automatically using a webhook.
+
+To do this, you'll use Builder's [Content API](https://www.builder.io/c/docs/content-api) and [Qwik API](https://www.builder.io/c/docs/qwik-api) to connect your Builder space to Astro with zero client-side JavaScript.
+
+:::tip[Official resource]
+If you are using a JavaScript framework (e.g. Svelte, Vue, or React) in your Astro project you can use one of Builder's integrations as an alternative to making raw fetch calls through the REST API.
+:::
+
+## Prerequisites
+
+To get started, you will need to have the following:
+
+* **A Builder account and space** - If you don't have an account yet, [sign up for free](https://www.builder.io/) and create a new space. If you already have a space with Builder, feel free to use it, but you will need to modify the code to match the model name and custom data fields.
+* **A Builder API key** - This public key will be used to fetch your content from Builder. [Read Builder's guide on how your key](https://www.builder.io/c/docs/using-your-api-key#finding-your-public-api-key).
+
+## Setting up credentials
+
+To add your Builder API key to Astro, create a `.env` file in the root of your project (if one does not already exist) and add the following variable:
+
+```ini title=".env"
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+```
```

Comment:
``````suggestion
To add your Builder API key and your Builder model name to Astro, create a `.env` file in the root of your project (if one does not already exist) and add the following variables:

```ini title=".env"
BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
BUILDER_BLOGPOST_MODEL='blogpost'
```
``````
So, since we're going to make having a project with a model called `blogpost` a prerequisite, we can get rid of that section below by adding to the `.env` file here.